pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -445,51 +540,123 @@ func (p *PoolManager) vmWaitingCleanupLook() {
 	for _ = range c {
 		p.poolMutex.Lock()
 
-		configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
-		if err != nil {
-			logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
-			p.poolMutex.Unlock()
+		p.handleStaleLegacyConfigMapEntries(logger)
+
+		p.handleStaleMacEntries(logger)
+
+		p.poolMutex.Unlock()
+	}
+}
+
+func (p *PoolManager) handleStaleMacEntries(parentLogger logr.Logger) error {
+	logger := parentLogger.WithName("handleStaleMacEntries")
+	var macMapChanged bool
+	for macAddress, macEntry := range p.macPoolMap {
+		macMapChanged = false
+		if !macEntry.hasPendingTransaction() {
 			continue
 		}
+		isEntryStale, err := p.isMacTransactionStale(macAddress)
+		if err == nil && isEntryStale {
+			macMapChanged = true
+			logger.Info("entry is stale", "macAddress", macAddress, "vmFullName", macEntry.instanceName, "interfaceName", macEntry.macInstanceKey, "stale TS", macEntry.transactionTimestamp)
+			vm, err := p.getvmInstance(macEntry.instanceName)
+			if err != nil && apierrors.IsNotFound(err) {
+				logger.Info("vm no longer exists or does not have Template. removing mac from pool", "macAddress", macAddress, "entry", macEntry)
+				p.macPoolMap.removeMacEntry(macAddress)
+			} else if err == nil {
+				p.macPoolMap.alignMacEntryAccordingToVmInterface(macAddress, getVirtualMachineInterfaces(vm))
+			} else {
+				return err
+			}
+		}
+	}
+	if macMapChanged {
+		logger.V(1).Info("macMap is updated", "macPoolMap", p.macPoolMap)
+	}
+	return nil
+}
 
-		configMapUpdateNeeded := false
-		if configMap.Data == nil {
-			logger.V(1).Info("the configMap is empty", "configMapName", names.WAITING_VMS_CONFIGMAP, "macPoolMap", p.macPoolMap)
-			p.poolMutex.Unlock()
+func (p *PoolManager) getvmInstance(vmFullName string) (*kubevirt.VirtualMachine, error) {
+	vmFullNameSplit := strings.Split(vmFullName, "/")
+	vmNamespace := vmFullNameSplit[1]
+	vmName := vmFullNameSplit[2]
```

Comment:
There is a type named `types.NamespacedName` that contains `Name` and `Namespace` field, so let's create this helper functions 

```golang
composeNamespacedName(string fullName) (types.NamespacedName, error)
getVMInstance(key types.NamespacedName).
```

I see other parts of the code that also do the Rest trick to retrieve the VM let's use this functions there too.

Reply:
I'm afraid it's not enough.. the macMap hold macs for both pods and vms.

so in there could be a pod and a vm with the same name & namespace, 
That's why I use `pod/<pod-namespace>/<pod-name>` and `vm/<vm-namespace>/<vm-name>` so the key is always unique